Run `python3 dev/update_changelog.py --prev-version 2.9...` (#10689)
mlflow/mlflow: CHANGELOG.md

# CHANGELOG

## 2.9.2 (2023-12-14)

MLflow 2.9.2 includes several major features and improvements

Features:

> - [] Create `mlflow.deployments.openai` (#10473, @prithvikannan)

Reviewer: ```suggestion
- [Deployments] Create `mlflow.deployments.openai` (#10473, @prithvikannan)
```

- [Server-infra] Add env var to disable redirects again (#10673, @daniellok-db)

> Bug fixes:

Reviewer: ```suggestion
Security fixes:
```

- [Server-infra] Add sandboxed jinja2 loader for yaml rendering (#10676, @BenWilson2)
- [Artifacts] Disable '..' in query string for artifact URI (#10653, @B-Step62)

> - [Server-infra / Windows] Fix for windows traversal attack (#10647, @BenWilson2)

Reviewer: ```suggestion
- [Windows] Fix for windows traversal attack (#10647, @BenWilson2)
```

# CHANGELOG

## 2.9.2 (2023-12-14)

> MLflow 2.9.2 includes several major features and improvements

Reviewer 2: ```suggestion
MLflow 2.9.2 is a patch release, containing several critical security fixes and configuration updates to support extremely large model artifacts. 
```

Features:

> - [Deployments] Add `mlflow.deployments.openai` (#10473, @prithvikannan)

Reviewer 2: ```suggestion
- [Deployments] Add the `mlflow.deployments.openai` API to simplify direct access to OpenAI services through the deployments API (#10473, @prithvikannan)
```

> - [Server-infra] Add env var to disable redirects again (#10673, @daniellok-db)

Reviewer 2: ```suggestion
- [Server-infra] Add a new environment variable that permits disabling http redirects within the Tracking Server for enhanced security in publicly accessible tracking server deployments (#10673, @daniellok-db)
```

> - [Artifacts] Make MPU / MPD chunk size configurable via environment variables (#10648, @harupy)

Reviewer 2: ```suggestion
- [Artifacts] Add a environment variable configurations for both Multi-part upload and Multi-part download that permits modifying the per-chunk size to support extremely large model artifacts (#10648, @harupy)
```

Security fixes:

> - [Server-infra] Add sandboxed Jinja loader for yaml rendering (#10676, @BenWilson2, #10640, @harupy)

Reviewer 2: ```suggestion
- [Server-infra] Disable the ability to inject malicious code via manipulated YAML files by forcing YAML rendering to be performed in a secure Sandboxed mode (#10676, @BenWilson2, #10640, @harupy)
```

> - [Artifacts] Disable `..` in query string for artifact URI (#10653, @B-Step62)

Reviewer 2: ```suggestion
- [Artifacts] Prevent path traversal attacks when querying artifact URI locations by disallowing `..` path traversal queries (#10653, @B-Step62)
```

- [Data] Fix for windows traversal attack in `HTTPDatasetSource` (#10647, @BenWilson2)

> - [Artifacts] Prevent path traversal with encoded URL (#10650, @B-Step62)

Reviewer 2: ```suggestion
- [Artifacts] Prevent a potential path traversal attack vector via encoded url traversal paths by decoding paths prior to evaluation  (#10650, @B-Step62)
```

> - [Artifacts] Use validated path after path validation (#1066, @harupy)

Reviewer 2: ```suggestion
- [Artifacts] Prevent the ability to conduct path traversal attacks by enforcing the use of sanitized paths with the tracking server (#10666, @harupy)
```

> - [Artifacts] Apply `os.path.basename` before `.` or `..` check in `FTPArtifactRepository` (#10657, @harupy)

Reviewer 2: ```suggestion
- [Artifacts] Prevent path traversal attacks when using an FTP server as a backend store by enforcing base path declarations prior to accessing user-supplied paths (#10657, @harupy)
```

Documentation updates:

> - [Docs] Adding Notebook for RAG Blog (#10661, @AbeOmor)

Reviewer 2: ```suggestion
- [Docs] Add an end-to-end tutorial for RAG creation and evaluation (#10661, @AbeOmor)
```

- [Docs] Add Tensorflow landing page (#10646, @chenmoneygithub)
- [Deployments / Tracking] Add endpoints to LLM evaluation docs (#10660, @prithvikannan)

> - [Examples] Add retriever evaluation tutorial and update question generation notebook (#10419, @liangz1)

Reviewer 2: ```suggestion
- [Examples] Add retriever evaluation tutorial for LangChain and improve the Question Generation tutorial notebook (#10419, @liangz1)
```
